grpc-proto/proto/query.proto

```diff
@@ -359,6 +359,9 @@ message Response {
   repeated string warnings = 2;
   // The tracing information, if it was requested for the query.
   Traces traces = 3;
+
+  // Denotes if the query is idempotent or not. If it is, it is safe to retry it client-side
+  bool is_idempotent = 5;
```

Comment:
I'm not sure how valuable adding an `is_idempotent` flag to the response is. A `Response` is the result of a successful execution and wouldn't require a retry. It more of value to add the the error response as a trailer.

Reply:
good point, I will propagate this info in the:
```
private <T> Metadata makeTrailer(Metadata.Key<T> key, T value) {
    Metadata trailer = new Metadata();
    trailer.put(key, value);
    return trailer;
  }
```
I wonder if there is a use-case for having that in the `Response` as well, or should we remove it.

Reply:
We decided that it is not safe to return that info to the client in any form.
The main problem with it is that the failure may happen before/during preparing logic - in that case, we will not have information about idempotency. The client-side retries will not be able to see the flag, as it will be not set. It will result in a convoluted logic, where it is safe to retry only if the `is_idempotent` is present and is `true`.
We will use is_idempotent only for the internal server-side retires.  (cc @mpenick)

Reply:
done in: https://github.com/stargate/stargate/pull/1194/commits/839d27cbf678d747146b5957577667e8fe171d4b